### PR TITLE
Restructured GitHub workflow and example app for web

### PR DESCRIPTION
**Pull request by mvanbeusekom**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Maintenance

### :arrow_heading_down: What is the current behavior?

Currently CI is configured to run one job configured with different steps to test format, analyse, etc... This has the disadvantage that steps are run synchronously and also hides tasks from the GitHub overview (as developer you'll have to do extra navigation to see which step failed).

Additionally the `geolocator_web` package doesn't have a dedicated example App which makes debugging more difficult. And also doesn't allow the CI to compile the changes.

### :new: What is the new behavior (if this is a feature change)?

The new configuration creates different jobs for each task solving the above problems and also gives more options for future configuration.

This PR also adds a dedicated example App for the `geolocator_web` package.

### :boom: Does this PR introduce a breaking change?

No

### :thinking: Checklist before submitting

- [x] I made sure all projects build.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [x] I updated the relevant documentation.
- [x] I rebased onto current `master`.

### CODE REVIEW

File: .github/workflows/app_facing_package.yaml [hunk 1]
```diff
@@ -16,13 +16,15 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # Job configuration for the geolocator (App Facing Package)
-  geolocator:
+  # Job configuration for the Flutter State Challenge App
```
Comment (renefloor) [inline .github/workflows/app_facing_package.yaml:19]: Multiple references to state challenge

File: .github/workflows/app_facing_package.yaml [hunk 2]
```diff
@@ -16,13 +16,15 @@ on:
...
+  format:
+    name: Format
+
     # The type of runner that the job will run on
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
```
Comment (renefloor) [inline .github/workflows/app_facing_package.yaml:24]: What's the benefit of Ubuntu?

**Review by renefloor — CHANGES_REQUESTED** [review]

File: geolocator_web/example/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+This project is a starting point for a Flutter application.
+
+A few resources to get you started if this is your first Flutter project:
+
+- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
+- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+
+For help getting started with Flutter, view our
+[online documentation](https://flutter.dev/docs), which offers tutorials,
+samples, guidance on mobile development, and a full API reference.
```
Comment (renefloor) [inline geolocator_web/example/README.md:16]: Would be nice to slightly update this

File: geolocator_web/example/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+            onPressed: () async {
+              await GeolocatorPlatform.instance.getCurrentPosition().then(
+                  (value) => {
+                        _positionItems.add(_PositionItem(
+                            _PositionItemType.position, value.toString()))
+                      });
+
+              setState(
+                () {},
+              );
```
Comment (renefloor) [inline geolocator_web/example/lib/main.dart:100]: I would set this on 1 line.
```suggestion
              setState(() {});
```

**Review by renefloor — CHANGES_REQUESTED** [review]

File: .github/workflows/app_facing_package.yaml (hunk 2 above)
Comment (mvanbeusekom) [reply .github/workflows/app_facing_package.yaml:24]: From a GitHub usage / license perspective there is no real benefit. GitHub runs their Windows and Ubuntu VMs in Azure (Standard DS2 v2 virtual machines). The macOS machines are part of GitHubs own macOS cloud solution. Personally I felt the Ubuntu configuration is more reliable but this is by no means backed by any proof. 

More information on GitHub Hosted runners can be found [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).

From a technical perspective I think it is good to also make sure things work on different operating systems.

File: .github/workflows/app_facing_package.yaml (hunk 1 above)
Comment (mvanbeusekom) [reply .github/workflows/app_facing_package.yaml:19]: Fixed.

File: geolocator_web/example/README.md (hunk 3 above)
Comment (mvanbeusekom) [reply geolocator_web/example/README.md:16]: Fixed.

File: geolocator_web/example/lib/main.dart (hunk 4 above)
Comment (mvanbeusekom) [reply geolocator_web/example/lib/main.dart:100]: Fixed.

File: geolocator_web/example/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+  _GeolocatorWidgetState createState() => _GeolocatorWidgetState();
+}
+
+class _GeolocatorWidgetState extends State<GeolocatorWidget> {
+  static final String _kLocationServicesDisabledMessage =
+      'Location services are disabled.';
+  static final String _kPermissionDeniedMessage = 'Permission denied.';
+  static final String _kPermissionDeniedForeverMessage =
+      'Permission denied forever.';
+  static final String _kPermissionGrantedMessage = 'Permission granted.';
```
Comment (renefloor) [inline geolocator_web/example/lib/main.dart:38]: Why wouldn't you define these as const?

I believe using 'kName' is not a convention in dart. [At least constant are just 'name'.](https://dart.dev/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names)

File: geolocator_web/example/lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+  final GeolocatorPlatform _geolocatorPlatform = GeolocatorPlatform.instance;
+  final List<_PositionItem> _positionItems = <_PositionItem>[];
+  StreamSubscription<Position>? _positionStreamSubscription;
+
+  @override
+  Widget build(BuildContext context) {
+    const sizedBox = SizedBox(
```
Comment (renefloor) [inline geolocator_web/example/lib/main.dart:46]: I would name this 'padding' or something like that.

**Review by renefloor — APPROVED** [review]
Some nits

**Pull request merged**